setup.sh

```diff
@@ -1,19 +1,50 @@
 #!/bin/bash
 
+# This script installs all the juju-1 and the dependencies.
+
+# Refresh the potentially stale apt cache.
 apt-get update -qq
+# Install software tools such as add-apt-repository.
 apt-get install -qy software-properties-common
+# Install the ppa for Juju and charm tools.
 apt-add-repository -y ppa:juju/stable
+# Refresh the apt-cache with the new Juju PPA.
 apt-get update -qq
 
-## don't install lxc & local till we get local support
-#apt-get install -qy lxc iptables juju-local
-
-## actually, we need juju-local since the local provider
-## can be used by bootstrapping from outside the container
-apt-get -qy install juju-local
-apt-get -qy install juju-core juju-quickstart juju-deployer python-jujuclient
-apt-get -qy install tmux charm-tools openssh-client
-apt-get -qy install virtualenvwrapper python-dev cython git
+# Install juju-local since the local provider can be used by bootstrapping 
+# from outside the container.
+apt-get install -qy \
+    byobu \
+    cython \
+    git \
+    juju-1 \
+    juju-core \
+    juju-deployer \
+    openssh-client \
+    python-dev \
+    sudo \
+    vim \
+    virtualenvwrapper
 
-useradd -m ubuntu
+# Add ubuntu to the passwordless sudo file.
 echo "ubuntu ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/juju-users
+JUJU=`which juju-1 > /dev/null && echo juju-1 || echo juju`
+# Get the version of Juju installed.
+JUJU_VERSION=`${JUJU} version`
+# The home directory for the ubuntu user.
+HOME=/home/ubuntu
+# Set some environment variables used by Juju.
+RC=${HOME}/.bashrc
```

Comment:
This is what I was talking about earlier.. You sure this is gonna work?  If JUJU_REPO is volume mounted over /home/ubuntu, will this .bashrc ever get run?